test/rekt/features/containersource/readyness.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package containersource
+
+import (
+	"fmt"
+
+	"knative.dev/eventing/test/rekt/features"
+	"knative.dev/eventing/test/rekt/resources/containersource"
+	"knative.dev/eventing/test/rekt/resources/svc"
+	"knative.dev/reconciler-test/pkg/feature"
+)
+
+// GoesReady returns a feature testing if a containersource becomes ready.
```

Comment:
I'm not sure I like this. Either the comment should be changed that this actually creates a ContainerSource and waits for it to be ready. I think it would be better to have something like (not real function names to protect the innocent).
Setup(containerSource.Install(...))
Feature(GoesReady())

Reply:
ok working on this

Reply:
ok please look now.